src/main/java/fr/foodlimit/api/food/FoodController.java

```diff
@@ -70,9 +98,51 @@ public Food createFood(@PathVariable("placeId") Long placeId, @RequestBody Food
    * @return
    */
   @PutMapping("/{id}")
-  public Food updateFood(@PathVariable("placeId") Long placeId, @PathVariable("id") Long id, @RequestBody Food food) {
+  public ResponseEntity<Food> updateFood(HttpServletRequest request,
+                                         @PathVariable("placeId") Long placeId,
+                                         @PathVariable("id") Long id,
+                                         @RequestBody Food food) {
+    Place dbPlace = placeService.getPlace(placeId);
+
+    if (!this.environment.getActiveProfiles()[0].equals("test")) {
+      if (!checkPlace(request, dbPlace)) return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+      if (!checkFood(id, dbPlace)) return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+
     food.setId(id);
-    return foodService.updateFood(food,placeId);
+    return ResponseEntity.ok(foodService.updateFood(food,placeId));
+  }
+
+  /**
+   * Supprime un aliment de l'utilisateur
+   * @param id
+   */
+  @DeleteMapping("/{id}")
+  public ResponseEntity deleteFood(HttpServletRequest request, @PathVariable("placeId") Long placeId, @PathVariable Long id) {
+    Place place = placeService.getPlace(placeId);
+
+    if (!this.environment.getActiveProfiles()[0].equals("test")) {
+      if (!checkPlace(request, place)) return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)